code/cfile/cfile.cpp

```diff
@@ -634,13 +635,14 @@ void cf_create_directory( int dir_type )
 
 	} while( current_dir != CF_TYPE_ROOT );
 
-	
 	int i;
 
 	for (i=num_dirs-1; i>=0; i-- )	{
 		cf_create_default_path_string( longname, sizeof(longname)-1, dir_tree[i], NULL );
-		mprintf(( "CFILE: Creating new directory '%s'\n", longname ));
-        mkdir_recursive(longname);
+		if( stat( longname, &statbuf ) != 0) {
```

Comment:
Ah, ok. If `stat` is available everywhere then that should work. I'm pretty sure that the path name isn't validated when the function arrives at the `stat`.